doc/source/whatsnew/v0.21.0.txt

```diff
@@ -79,6 +79,62 @@ Other Enhancements
 - :func:`date_range` now accepts 'YS' in addition to 'AS' as an alias for start of year (:issue:`9313`)
 - :func:`date_range` now accepts 'Y' in addition to 'A' as an alias for end of year (:issue:`9313`)
 
+.. _whatsnew_0210.enhancements.read_csv:
+
+``read_csv`` use `python-requests` (if installed) to support basic auth and much more
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If `python-requests` library is installed try to use it first. If not, continue using urllib
```

Comment:
The reasoning is that we want our code to be backwards compatible, so we don't want to introduce any inconsistencies by using a new library to handle requests for the time being.

In the future, we *may* transition to using `requests` entirely (the interface is definitely better, I won't deny that), but such transitions are gradual so that people have to adjust their code bases that depend on `pandas`.